docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/LoadImages.cs

```diff
@@ -41,29 +42,39 @@ public static void Example()
             var pipeline = mlContext.Transforms.LoadImages("ImageObject", imagesFolder, "ImagePath");
 
             var transformedData = pipeline.Fit(data).Transform(data);
-            // The transformedData IDataView contains the loaded images now.
 
+            PrintColumns(transformedData);
             // Preview the transformedData. 
-            var transformedDataPreview = transformedData.Preview();
-            PrintPreview(transformedDataPreview);
             // ImagePath    Name         ImageObject           
-            // tomato.bmp   tomato       System.Drawing.Bitmap
-            // banana.jpg   banana       System.Drawing.Bitmap
-            // hotdog.jpg   hotdog       System.Drawing.Bitmap
-            // tomato.jpg   tomato       System.Drawing.Bitmap
+            // tomato.bmp   tomato       {Width=800, Height=534}
+            // banana.jpg   banana       {Width=800, Height=288}
+            // hotdog.jpg   hotdog       {Width=800, Height=391}
+            // tomato.jpg   tomato       {Width=800, Height=534}
         }
 
-        private static void PrintPreview(DataDebuggerPreview data)
+        private static void PrintColumns(IDataView transformedData)
         {
-            foreach (var colInfo in data.ColumnView)
-                Console.Write("{0,-25}", colInfo.Column.Name);
-
-            Console.WriteLine();
-            foreach (var row in data.RowView)
+            // The transformedData IDataView contains the loaded images now.
+            Console.WriteLine("{0, -25} {1, -25} {2, -25}", "ImagePath", "Name", "ImageObject");
+            using (var cursor = transformedData.GetRowCursor(transformedData.Schema))
             {
-                foreach (var kvPair in row.Values)
-                    Console.Write("{0,-25}", kvPair.Value);
-                Console.WriteLine();
+                var imagePathGetter = cursor.GetGetter<ReadOnlyMemory<char>>(cursor.Schema["ImagePath"]);
+                var nameGetter = cursor.GetGetter<ReadOnlyMemory<char>>(cursor.Schema["Name"]);
+                var imageObjectGetter = cursor.GetGetter<Bitmap>(cursor.Schema["ImageObject"]);
+                while (cursor.MoveNext())
+                {
+                    ReadOnlyMemory<char> imagePath = null;
```

Comment:
>null [](start = 53, length = 4)

Do not do `ReadOnlyMemory<char> imagePath = null`, prefer `ReadOnlyMemory<char> imagePath = default`. There is no need to invoke the implicit conversion since it can be so easily avoided.

Reply:
In addition of course to the earlier advice to put this declaration *outside* the loop, but of course that advice applies everywhere.

---
In reply to: [275877068](https://github.com/dotnet/machinelearning/pull/3307#discussion_r275877068) [](ancestors = 275877068)

Reply:
Good idea.